### ISSUE TITLE
Default viewer doesn't use 'retina' toolbar icons on iOS

### ISSUE THREAD

**Issue by jasongisstl** [mozilla/pdf.js#6685]
When using the default viewer provided with PDF.js on iOS (retina), the toolbar icons appear blurry.

This seems to be because the following media query in _viewer.css_ isn't being matched:
`@media screen and (min-resolution: 2dppx)`

Changing this to include a webkit-specific prefix seems to solve this:
`@media screen and (min-resolution: 2dppx), (-webkit-min-device-pixel-ratio: 2)`
(My CSS knowledge isn't great, so apologies if there are problems with this)

I realise that users are expected to build upon the default viewer, but it would be nice if this worked out of the box.

(Tested using iOS 9.1 Safari & UIWebview)

**Comment by yurydelendik** [comment]
Safari on desktop affected as well, the following shall be fine for our preprocessor:

```
@media
  screen and (-webkit-min-device-pixel-ratio: 2),
  screen and (min-resolution: 2dppx) {
```

**Comment by prohtex** [comment]
I'm wondering if any progress has been made on this one. I am implementing pdfjs for clients who exclusively use tablets and mobile devices, and many comment that the icons are "blurry." On a desktop with a high-resolution display, the issue is not as critical as the icons are much smaller relative to the view area. But on a mobile device, it is odd.

![img_0829](https://user-images.githubusercontent.com/12946938/35244375-c340a3f0-ff74-11e7-8cbb-dff7ec4970be.PNG)

**Comment by yurydelendik** [comment]
> I'm wondering if any progress has been made on this one.

Nope. Any ideas why mobile safari is doing it?

**Comment by prohtex** [comment]
Isn't the issue that no one has created high-resolution assets to be served to the browser? I'm also wondering why the desktop viewer always displays a sans-serif font while iOS uses what looks like Times Roman!

**Comment by yurydelendik** [comment]
> Isn't the issue that no one has created high-resolution assets to be served to the browser? 

Are these need to be in Safari specific format? WFM on Firefox and Chorme. Also text looks good as well.

**Comment by prohtex** [comment]
I'm using Safari mobile on iOS devices. I don't have access to an android phone rn to check, but I assumed the assets simply weren't large enough to display properly.

**Comment by timvandermeij** [comment]
Every image has a `@2x` variant for hi-dpi screens and this is served when the media query matches, so from our perspective this should be working (and has been tested on actual hi-dpi screens). Most likely Safari mobile on iOS is doing something strange that causes the media query not to match.

**Comment by prohtex** [comment]
Thank you @timvandermeij for this information. I'm not sure how to troubleshoot this, but can report that I've viewed many builds of pdfjs over the past 3 years on a broad array of iOS devices -- iPhones and iPads of all stripes. In every case, the wrong assets were served.

**Comment by timvandermeij** [comment]
It looks like the problem is that Safari only supports the `webkit` prefixed versions of the device pixel media queries. Referring to https://caniuse.com/#feat=css-media-resolution, it looks like Safari is the only one that requires a prefixed media query for this, so that's rather unfortunate. However, the solution mentioned in the original message above would most likely solve this. It would be good if you could test if adding `-webkit-min-device-pixel-ratio: 2` solves the issue. If so, we could find a way to incorporate it in PDF.js.

**Comment by Snuffleupagus** [comment]
> I'm not sure how to troubleshoot this,

Did you try the what's suggested in https://github.com/mozilla/pdf.js/issues/6685#issuecomment-159354026, since that seems like a good starting point?

**Comment by prohtex** [comment]
This issue also impacts ALL versions of Safari on Macs, as seen below

![screen shot 2018-02-01 at 9 52 23 pm copy](https://user-images.githubusercontent.com/12946938/35718723-a95317ba-079a-11e8-80ea-f094a171e34c.jpg)

**Comment by prohtex** [comment]
I tried sticking this code in a variety of places without any success. It can be seen currently here in viewer.css. Any ideas?

http://www.thing.gallery/test/pdf.js/web/viewer.html

**Comment by prohtex** [comment]
I can confirm that the solution suggested does work, on Safari and iOS. All that needs to be changed is this line in viewer.css:

	@media screen and (min-resolution: 2dppx) {

to

	@media
		screen and (-webkit-min-device-pixel-ratio: 2),
		screen and (min-resolution: 2dppx) {

This can be seen working here:

http://www.thing.gallery/test/pdf.js/web/viewer.html

**Comment by prohtex** [comment]
@timvandermeij, it seems the solution I have posted above completely resolves this issue. I am not exactly sure how to create a PR for this, but was wondering if one of the devs might be able to create the PR given it is only 2 lines of code. I tried just now but I don't think I have the permissions.

**Comment by darrenahunter** [comment]
this is a big +1 from me too
thanks

**Comment by timvandermeij** [comment]
Closing since this is fixed by #9629.
